src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -185,6 +188,11 @@ protected override async Task RemoveKeyAsync(string cacheKey, CacheSerializerHin
                         Logger.BackPropagateL2toL1(_logger, memoryCacheEntryOptions.Size ?? 0);
                         _memoryCache.Set(cacheKey, result, memoryCacheEntryOptions);
                         Logger.MemoryCacheCount(_logger, _memoryCacheType, read, _memoryCache.Count);
+
+                        if (telemetryData != null)
+                        {
+                            telemetryData.CacheLevel = Client.Cache.CacheLevel.L2Cache;
```

Comment:
I put it here because it is in the null check for the result. I figured if you did not get a value from the L2 cache it doesn't really count as a cache hit. Or does it?

I guess I should move the logic out of the null check for the memory cache.